### DISCUSSION TITLE
Update Dataverse from Power Platform (i.e. Power Apps/Power Automate)

### DISCUSSION THREAD

**Discussion by TPSQL23** [MarkMpn/Sql4Cds#311] [Q&A]
Please let me know of a way to trigger a stored procedure, from a Power Automate flow, that will update a Dataverse table?

For context, there is an existing stored procedure in a SQL Database.
I would like to switch from using SQL Database to using Dataverse (to bypass performance, gateway, private endpoint limitations).
The stored procedure contains advanced sql queries.
I would therefore like to use the SQL 4 CDS tool to create a stored procedure for the new Dataverse table.
I do not yet know how to fetch/call/trigger that stored procedure from the Power Automate flow. 
(The Power Automate flow currently calls a SQL connector to execute the stored procedure but that connector comes with limitations - the reason for this question).  

Please let me know of a way to trigger a stored procedure (without the SQL connector), from a Power Automate flow, that will update a Dataverse table?

**Comment by MarkMpn** [comment]
There isn’t a direct equivalent of a stored procedure in Dataverse. You could look at building a [Custom API](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/custom-api) to replace your stored procedure logic and you’d be able to execute that from Power Automate, but this wouldn’t involve SQL 4 CDS at all.

**Reply by TPSQL23** [reply]
Thank you for taking the time to respond
